remote/remote_state_s3.go

```diff
@@ -139,6 +140,9 @@ func validateS3Config(config *RemoteStateConfigS3, terragruntOptions *options.Te
 // confirms, create the bucket and enable versioning for it.
 func createS3BucketIfNecessary(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
 	if !DoesS3BucketExist(s3Client, config) {
+		if !shell.CanAcceptPrompt() {
+			return fmt.Errorf("remote: S3 bucket does not exist, or you don't have permissions to access it: %q", config.Bucket)
+		}
```

Comment:
A couple lines below this, we call `shell.PromptUserForYesNo`. I think all the user prompts go through that `shell` package. It may make more sense to put this check into that method, which, BTW, also handles the `--terragrunt-non-interactive` flag, which has the opposite behavior (assumes "yes" for all prompts).